### PR TITLE
Add a generate extracted text service following the established service pattern

### PR DESCRIPTION
**Pull request by elrayle**

### CODE REVIEW

File: spec/hydra/works/services/persist_derivatives_spec.rb [hunk 1]
```diff
@@ -88,5 +88,21 @@
         expect(generic_file.thumbnail.mime_type).to eq('image/jpeg')
       end
     end
+
+    context 'with a pdf file' do
```
Comment (jcoyne) [inline spec/hydra/works/services/persist_derivatives_spec.rb]: I don't understand how this fits with the rest of this pull request. Was this supposed to go here?

File: lib/hydra/works/services/generic_file/generate_extracted_text.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module Hydra::Works
+  class GenerateExtractedText
+    def self.call(object, content: :original_file)
+      fail ArgumentError, "object has no content at #{content} from which to generate extracted text" if object.send(content).nil?
+
+      # Always replace the thumbnail with whatever is from the original file
```
Comment (jcoyne) [inline lib/hydra/works/services/generic_file/generate_extracted_text.rb]: Thumbnail?

File: spec/hydra/works/services/generic_file/generate/extracted_text_spec.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
+require 'spec_helper'
+
+describe Hydra::Works::GenerateExtractedText do
+  context 'when the object has no original file' do
+    let(:error_message) { 'object has no content at original_file from which to generate extracted text' }
+    let(:object) { double('object', original_file: nil) }
+    it 'raises an error' do
+      expect(-> { described_class.call(object) }).to raise_error(ArgumentError, error_message)
```
Comment (jcoyne) [inline spec/hydra/works/services/generic_file/generate/extracted_text_spec.rb]: `expect { described_class.call(object) }.to raise_error...` is the more common way of doing this.

File: spec/hydra/works/services/generic_file/generate/extracted_text_spec.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+require 'spec_helper'
+
+describe Hydra::Works::GenerateExtractedText do
```
Comment (jcoyne) [inline spec/hydra/works/services/generic_file/generate/extracted_text_spec.rb]: This test should probably have a mock for FullTextExtractionService and not call it.

**Comment by jcoyne** [comment]
Should we close this now that text extraction has moved to hydra-derivatives?

**Comment by mjgiarlo** [comment]
I'm ok with it if @elrayle is.

**Pull request closed without merging**